rosbag2_compression/src/rosbag2_compression/zstd_decompressor.cpp

```diff
@@ -156,6 +162,41 @@ void write_output_buffer(
   fclose(file_pointer);
 }
 
+/**
+ * Throws a runtime_error if data does not contain a ZSTD compressed frame.
+ */
+void throw_on_invalid_magic_number(const std::vector<uint8_t> & data)
+{
+  if (data.size() < kMagicNumberSize) {
+    std::stringstream errmsg;
+    errmsg << "Failed to process data; data frame size: " <<
+      data.size() << " bytes is too small to contain a ZSTD frame. " <<
+      "Required size is at least " << kMagicNumberSize << " bytes.";
+
+    throw std::runtime_error{errmsg.str()};
+  }
+
+  uint32_t magic_number = 0;
+  auto ptr_magic_number = reinterpret_cast<uint8_t *>(&magic_number);
+  for (int i = 0; i < static_cast<int>(kMagicNumberSize); ++i) {
+    ptr_magic_number[i] = data[i];
+  }
```

Comment:
Why not reinterpret_cast to uint32_t directly to save the loop? (I may be missing something here).